pgjdbc/src/main/java/org/postgresql/util/PGtokenizer.java

```diff
@@ -37,71 +49,86 @@ public PGtokenizer(String string, char delim) {
     tokenize(string, delim);
   }
 
-  /**
-   * This resets this tokenizer with a new string and/or delimiter.
-   *
-   * @param string containing tokens
-   * @param delim single character to split the tokens
-   * @return number of tokens
-   */
-  public int tokenize(String string, char delim) {
-    tokens.clear();
-
-    // nest holds how many levels we are in the current token.
-    // if this is > 0 then we don't split a token when delim is matched.
-    //
-    // The Geometric datatypes use this, because often a type may have others
-    // (usualls PGpoint) imbedded within a token.
-    //
-    // Peter 1998 Jan 6 - Added < and > to the nesting rules
-    int nest = 0;
-    int p;
-    int s;
-    boolean skipChar = false;
-    boolean nestedDoubleQuote = false;
-    char c = (char)0;
-    for (p = 0, s = 0; p < string.length(); p++) {
-      c = string.charAt(p);
-
-      // increase nesting if an open character is found
-      if (c == '(' || c == '[' || c == '<' || (!nestedDoubleQuote && !skipChar && c == '"')) {
-        nest++;
-        if (c == '"') {
-          nestedDoubleQuote = true;
-          skipChar = true;
-        }
-      }
-
-      // decrease nesting if a close character is found
-      if (c == ')' || c == ']' || c == '>' || (nestedDoubleQuote && !skipChar && c == '"')) {
-        nest--;
-        if (c == '"') {
-          nestedDoubleQuote = false;
-        }
-      }
-
-      skipChar = c == '\\';
-
-      if (nest == 0 && c == delim) {
-        tokens.add(string.substring(s, p));
-        s = p + 1; // +1 to skip the delimiter
-      }
+	/**
+	 * This resets this tokenizer with a new string and/or delimiter.
+	 *
+	 * @param string containing tokens
+	 * @param delim single character to split the tokens
+	 * @return number of tokens
+	 */
+	public int tokenize(String string, char delim) {
+		tokens.clear();
 
-    }
+		final Stack<Character> stack = new Stack<>();
```

Comment:
declare member variables prior to the constructor.
also, prefer use of `java.util.Deque` and `java.util.ArrayDeque` over the use of `java.util.Stack` (this is basically the same as preferring use of a List/ArrayList over Vector)

Reply:
stack is not a member  
java.util.Deque is indeed better in this case